codegen/aws/core/src/main/java/software/amazon/smithy/python/aws/codegen/AwsMkDocsFileGenerator.java

```diff
@@ -0,0 +1,525 @@
+                    fileWriter.write("### Event Stream Structure\n\n");
+                    fileWriter.write("#### Output Event Type\n\n");
+                    fileWriter.write("[`$L`](../unions/$L.md)\n\n",
+                            streamSymbol.getName(),
+                            streamSymbol.getName());
+                    fileWriter.write("### Initial Response Structure\n\n");
+                }
+
+                fileWriter.write("::: " + outputSymbol.toString());
+                // Use heading level 4 for event streams, level 3 for regular operations
+                int headingLevel = (hasInputStream || hasOutputStream) ? 4 : 3;
+                fileWriter.write("""
+                            options:
+                                heading_level: $L
+                        """, headingLevel);
+
+                // Add Errors documentation
+                var operationErrors = operation.getErrorsSet();
+                if (!operationErrors.isEmpty()) {
+                    fileWriter.write("\n## Errors\n\n");
+                    for (var errorId : operationErrors) {
+                        var errorShape = context.model().expectShape(errorId);
+                        var errorSymbol = context.symbolProvider().toSymbol(errorShape);
+                        String errorName = errorSymbol.getName();
+                        fileWriter.write("- [`$L`](../errors/$L.md)\n", errorName, errorName);
+                    }
+                }
+            });
+        }
+    }
+
+    private static final class StructureGenerationInterceptor
+            implements CodeInterceptor.Appender<StructureSection, PythonWriter> {
+
+        private final GenerationContext context;
+        private final TreeSet<String> structures;
+
+        public StructureGenerationInterceptor(GenerationContext context, TreeSet<String> structures) {
+            this.context = context;
+            this.structures = structures;
+        }
+
+        @Override
+        public Class<StructureSection> sectionType() {
+            return StructureSection.class;
+        }
+
+        @Override
+        public void append(PythonWriter pythonWriter, StructureSection section) {
+            var shape = section.structure();
+            var symbol = context.symbolProvider().toSymbol(shape);
+            String symbolName = symbol.getName();
+            String docsFileName = String.format("docs/structures/%s.md", symbolName);
+            // Don't generate separate docs for input/output structures (they're included with operations)
+            if (!shape.hasTrait(InputTrait.class) && !shape.hasTrait(OutputTrait.class)) {
+                structures.add(symbolName);
+                context.writerDelegator().useFileWriter(docsFileName, "", writer -> {
+                    writer.write("::: " + symbol.toString());
+                    writer.write("""
+                                options:
+                                    heading_level: 1
```

Comment:
Can we change the default heading_level to 1 so we don't need to specify this case? Then if we need to go lower, we can be explicit

Related Docs: https://mkdocstrings.github.io/python/usage/configuration/headings/#heading_level

Reply:
This comment isn't relevant anymore since the .md files aren't going to be generated statically during codegen.